src/scraper/spider_manager.py

```diff
@@ -0,0 +1,81 @@
+# external imports
+import scrapy
+from scrapy.crawler import CrawlerProcess
+import scrapy.signals
+
+# Project imports
+import scraper.scrapy_settings as settings
+
+# Python imports
+from typing import List
+
+
+class SpiderManager:
```

Comment:
this class will need some unit tests too :)

Reply:
I'll have a look at how to test this, given that it has the CrawlerProcess + spider things. 
@m4r4c00ch0 maybe you would have some input here?

Reply:
What's exactly the purpose of this class? I don't like the word Manager because it looks like this module might become a _God Object_. But, I might be wrong. Is this a [Mediator](http://en.wikipedia.org/wiki/Mediator_pattern)?

p.s. I am used to writing tests before programming, so this might need refactoring before we can test it easily. I would give you my advice on how to unit test this class, but first I need to know its purpose.

Reply:
It's purpose is to implement scraping and parsing functions for spiders, so they are not directly managed by a BaseScraper subclass. This way, BaseScrapyScraper can just delegate that behavior on the manager object. Maybe it's better to merge that logic into BaseScrapyScraper? Since the latter is just redirecting behavior into SpiderManager.